hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/NaiveStreamingTreeHasher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.junit.support.validators.block;
+
+import static com.hedera.node.app.hapi.utils.CommonUtils.noThrowSha384HashOf;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.blocks.StreamingTreeHasher;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+
+public class NaiveStreamingTreeHasher implements StreamingTreeHasher {
+    private static final byte[] EMPTY_HASH = noThrowSha384HashOf(new byte[0]);
+
+    private final List<Bytes> leaves = new ArrayList<>();
+    private boolean rootHashRequested = false;
+
+    public static Bytes hashNaively(@NonNull final List<Bytes> leaves) {
+        final var hasher = new NaiveStreamingTreeHasher();
+        for (final var item : leaves) {
+            hasher.addLeaf(item);
+        }
+        return hasher.rootHash().join();
+    }
+
+    @Override
+    public void addLeaf(@NonNull final Bytes leaf) {
+        if (rootHashRequested) {
```

Comment:
Just checking, `rootHashRequested` doesn't need to be thread-safe, right? Seems like concurrent calls to `addLeaf` _after_ `rootHash()` would be possible in some situations, but I doubt `StateChangesValidator` uses multiple threads

Reply:
No This is already used class in JMH benchmarks. So used the same.